stdlib/public/core/StringGuts.swift

```diff
@@ -51,10 +51,22 @@ extension _StringGuts {
   }
 
   @inlinable @inline(__always)
-  internal init(_ bufPtr: UnsafeBufferPointer<UInt8>, isASCII: Bool) {
+  internal init(immortal bufPtr: UnsafeBufferPointer<UInt8>, isASCII: Bool) {
     self.init(_StringObject(immortal: bufPtr, isASCII: isASCII))
   }
 
+  @usableFromInline
+  internal init(_ bufPtr: UnsafeBufferPointer<UInt8>, isASCII: Bool) {
```

Comment:
Do you know where this is being called from? I want to make sure this change doesn't accidentally affect callers we're not sure of. You could consider adding a label `copying` or some such to make it clearer what will happen.

Reply:
I've moved this initializer into LegacyABI.swift (with its original body).